comp/metadata/runner/runnerimpl/runner.go

```diff
@@ -72,11 +92,30 @@ func NewEmptyProvider() Provider {
 
 // createRunner instantiates a runner object
 func createRunner(deps dependencies) *runnerImpl {
+	providers := []MetadataProvider{}
+	priorityProviders := []PriorityMetadataProvider{}
+
+	nonNilProviders := fxutil.GetAndFilterGroup(deps.Providers)
+	nonNilPriorityProviders := fxutil.GetAndFilterGroup(deps.PriorityProviders)
+
+	for _, optionaP := range nonNilProviders {
+		if p, isSet := optionaP.Get(); isSet {
+			providers = append(providers, p)
+		}
+	}
+
+	for _, optionaP := range nonNilPriorityProviders {
+		if p, isSet := optionaP.Get(); isSet {
+			priorityProviders = append(priorityProviders, p)
+		}
+	}
```

Comment:
I moved this functionality here mostly to leave the `start` function clean and to report correct number of providers.

The log line `r.log.Debugf("Starting metadata runner with %d providers", len(r.providers)+len(r.priorityProviders))` was reporting wrong numbers as it was also counting optional providers